packages/@react-aria/datepicker/src/useDateSegment.ts

```diff
@@ -379,7 +362,16 @@ export function useDateSegment(segment: DateSegment, state: DateFieldState, ref:
       onKeyDown,
       onFocus,
       style: {
-        caretColor: 'transparent'
+        caretColor: 'transparent',
+        userSelect: 'none',
+        WebkitUserSelect: 'none'
```

Comment:
Ugh, no I'll look into it.

Reply:
hmm I can't reproduce the text selection issue on my android phone

Reply:
Did some more digging and it turns out the press and hold behavior I'm seeing existed prior to the changes in this PR. Just to be clear, this is what happens when I press and hold on a date segment (aka the context menu w/ "Paste"/"Select All")
![Screenshot_20220413-135613_Chrome](https://user-images.githubusercontent.com/9661127/163269253-4662f9eb-0e34-495d-96cb-68892e6d4d0e.jpg)

I had assumed that this context menu shouldn't appear at all due to "user-select: none" since this is how it behaves on iOS. Not the biggest issue so I'm happy to punt/wont-fix this

Reply:
Hmm not really sure if there's anything we can do about that one without going full usePress on it.